inc/class-wpseo-shortlinker.php

```diff
@@ -58,4 +57,56 @@ public static function get( $url ) {
 	public static function show( $url ) {
 		echo esc_url( self::get( $url ) );
 	}
+
+	/**
+	 * Gets the current site's PHP version, without the extra info.
+	 *
+	 * @return string The PHP version.
+	 */
+	private function get_php_version() {
+		$version = explode( '.', PHP_VERSION );
+
+		return $version[0] . '.' . $version[1] . '.' . $version[2];
+	}
+
+	/**
+	 * Get our software and whether it's active or not.
+	 *
+	 * @return string The software name + activation state.
+	 */
+	private function get_software() {
+		$software = 'free';
+		if ( class_exists( 'WPSEO_Product_Premium' ) ) {
+			$software          = 'premium-inactive';
+			$product_premium   = new WPSEO_Product_Premium();
+			$extension_manager = new WPSEO_Extension_Manager();
+
+			if ( $extension_manager->is_activated( $product_premium->get_slug() ) ) {
+				$software = 'premium-activated';
+			}
+		}
+
+		return $software;
+	}
+
+	/**
+	 * Gets the current user's role without leaking roles that shouldn't be public.
+	 *
+	 * @return string The filtered user role.
+	 */
+	private function get_filtered_user_role() {
+		$user           = wp_get_current_user();
+		$built_in_roles = array(
+			'administrator',
+			'wpseo_manager',
+			'wpseo_editor',
+			'editor',
+			'author',
+			'contributor',
+			'subscriber',
+		);
+		$filtered_roles = array_intersect( $built_in_roles, $user->roles );
+
+		return current( $filtered_roles );
```

Comment:
If the user doesn't have a role, the result of this might be `false`. Should we catch that?